### ISSUE TITLE
Adaptive-FISTA

### ISSUE THREAD

**Issue by mrava87** [PyLops/pylops#323]
## Feature

Recently an Adaptive-FISTA algorithm was introduced in https://www.earthdoc.org/content/papers/10.3997/2214-4609.202229010. This algorithm shows faster convergence for problems where the model is expected to be very sparse (as it approximates the `L0` norm).

### Implementation

@NickLuiken will implement this in `pylops.optimization.sparsity.ISTA` and `pylops.optimization.sparsity.FISTA`

**Comment by cako** [comment]
Looks very promising. Would be interested to see if this means we can relax having to know the maximum eigenvalue. Maybe setting a sufficiently large value is enough under this adaptive scheme. Looking forward to seeing this!

**Comment by mrava87** [comment]
Good point! We haven’t looked into this but definitely for some problem where you can’t tell upfront the eigenvalue, not having to know it with high accuracy would be a bit plus :)

**Comment by cako** [comment]
Now that #90 is closed, it would be interesting to convert the AFISTA code into a class-based solver:
https://github.com/PyLops/pylops_transform2022/blob/main/afista.py

@mrava87 @NickLuiken

**Comment by rohanbabbar04** [comment]
Hi @cako @mrava87, If this issue is not touched upon, would like to work on it.

**Comment by mrava87** [comment]
Hi, this issue is already taken, we just need to move codes around so no point for you to look into it.
